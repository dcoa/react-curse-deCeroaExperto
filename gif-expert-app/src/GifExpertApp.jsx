import { useState } from 'react'
import { AddCategory } from './componets/AddCategory';
import { GifGrid } from './componets/GifGrid';

const GifExpertApp = props => {
    //const category = ['One Punch', 'Samurai X', 'Dragon Ball', 'One Piece', 'Gintama'];
    const [categories, setCategories] = useState(['Gintama']);

    // const handleApp =() => {
    //     setCategories([...categories, "Hunter x Hunter"])
    // }
    return (
        <>
            <h2>Gif Expert App</h2>
            <AddCategory setCategory={setCategories} /> 
            <hr />
            <ul>
                {
                    categories.map(anime => <GifGrid key={anime} Category={anime}>{anime}</GifGrid>)
                }
            </ul>
        </>
    )
}

export default GifExpertApp;
