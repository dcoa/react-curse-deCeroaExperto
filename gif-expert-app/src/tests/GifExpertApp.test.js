import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from './../GifExpertApp'

describe('Test in GifExpertApp', () => {
    const wrapper = shallow(<GifExpertApp/>)

    test('should render on match with snapshot', () => {
        expect(wrapper).toMatchSnapshot();

    })
    test('should show 1 GifGrid at initial state', () => {
        expect(wrapper.find('GifGrid').length).toBe(1)
    })
    
})
