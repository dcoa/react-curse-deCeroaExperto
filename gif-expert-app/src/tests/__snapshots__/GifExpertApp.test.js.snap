// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifExpertApp should render on match with snapshot 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      Category="Gintama"
      key="Gintama"
    >
      Gintama
    </GifGrid>
  </ul>
</Fragment>
`;
